compiler/src/lowering/mappers/types.rs

```diff
@@ -0,0 +1,16 @@
+use crate::lowering::names;
+use fe_analyzer::namespace::types::Type;
+use fe_analyzer::Context;
+use fe_parser::ast as fe;
+use fe_parser::node::Node;
+
+pub fn type_desc(context: &Context, desc: Node<fe::TypeDesc>) -> Node<fe::TypeDesc> {
+    let typ = context.get_type_desc(&desc).expect("missing attributes");
+
+    match typ {
+        Type::Tuple(tuple) if !tuple.is_empty() => {
```

Comment:
I think we should allow to define zero sized structs probably just as `struct Something. That's `struct` followed by the name and then not followed by a `:`. Rust has support for empty structs, too so we are in good company.

Reply:
Not saying that has to be in this PR so. But worth creating an issue for.

Reply:
This is what I was thinking too :+1: